### DISCUSSION TITLE
Web UI not accessible / Faulty DB setup during fresh install

### DISCUSSION THREAD

**Discussion by uz-ext-r** [apache/airflow#25131] [General] [answered]
### Apache Airflow version

2.3.3 (latest released)

### What happened

Fresh install of 2.3.3 DB on both MySQL and PostgreSQL caused `Your user has no roles and/or permissions!` error on Web UI.

### What you think should happen instead

Web UI should have been viewed as usual after login.

### How to reproduce

Steps:
1. Make a fresh install of the DB from scratch with version 2.3.3.
2. Create a user with admin role from the CLI as usual.
3. Try to login from the Webserver using the Admin account just created. 
4. An error page will appear, that states; `Your user has no roles and/or permissions!` although the user is definitely assigned the Admin role which is verified by running  `airflow users list` clearly showing `Admin` under the `roles` column of the user.

Additional info:
* The command `airflow roles list` will only result in the role `Admin` being listed. Other roles `Op`, `Public`, `User`, `Viewer` won't appear.
 * Creating a user with a role other than `Admin` isn't possible as the other roles don't exist.
* None of the issues above will occur on version 2.3.2. 

Workaround:
* Initialize DB with the version 2.3.2 and upgrade to 2.3.3.

### Operating System

Docker image python:3.8.13-buster running on K8S

### Versions of Apache Airflow Providers

apache-airflow-providers-google = "7.0.0"
apache-airflow-providers-mysql = "^2.2.3"
apache-airflow-providers-ssh = "^2.4.4"
apache-airflow-providers-cncf-kubernetes = "4.0.2"
apache-airflow-providers-amazon = "3.4.0"

### Deployment

Other Docker-based deployment

### Deployment details

Kubernetes Deployment.

### Anything else

Only when doing a complete fresh install from 2.3.3

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
You must have made some mistke. 

I run clean airlfow 2.3.3 with clean DB. Run ` airflow users create -u admin -p admin -f Thor -l Adminstra -r Admin -e dummy@dummy.email` and I could login without any problem. 

All looks good:

![image](https://user-images.githubusercontent.com/595491/179569397-86691b24-dc6b-4300-aae2-9358be45ed7b.png)

**Reply by uz-ext-r** [reply]
I can't say for certain but then this might be configuration dependent. In my case, the problems certainly go away when I initialize the DB with 2.3.2.

**Reply by potiuk** [reply]
Then you must do something else than you described.

What special configuration/initialization do you have or run?

There is no way anyone can help with the problem of your configuration if you don't look at it or provide a reproducible case which can be well, reproduced. I tried to do what you did and faied to reproduce it.

Can you please try to get "clear installation" (new virtualenv, AIRFLOW_HOME clean machine) following my steps, without any extra steps or configuration and see if it works and gradually add your configuration (or whatever else you do) to reproduce it - without it, there is not much help for anyone helping you.

**Comment by uz-ext-r** [comment, marked as answer]
Just wanted to update on this, although it is a bit too late. 

In the Python installation of our docker image, we're also installing a library called great-expectations, which had a dependency to an older version of the Jinja which was apparently in conflict with the version Airflow uses.

After we upgraded our great-expectations version from 0.13.x to 0.15.x the problem was resolved. Thanks for your help @potiuk.

**Reply by potiuk** [reply]
Thanks for reporting the finding. This would be really useful for someone who might stumble on similar issue !

**Comment by jfallt** [comment]
We had this same issue when upgrading to 2.9.2 

We fixed by removing any old images/volumes
`docker-compose down`
`docker-compose down --volumes --remove-orphans`
`docker-compose down --volumes --rmi all`
`docker system prune -a`

and in docker-compose.yaml, change
AIRFLOW__WEBSERVER__UPDATE_FAB_PERMS: 'true'
